cpp/src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITValue.cpp

```diff
@@ -39,16 +39,19 @@ JITValue& LLVMJITValue::assign(JITValue& value) {
 }
 
 JITValuePointer LLVMJITValue::andOp(JITValue& rh) {
+  CHECK(rh.getValueTypeTag() == getValueTypeTag());
+
   LLVMJITValue& llvm_rh = static_cast<LLVMJITValue&>(rh);
   llvm::Value* ans = nullptr;
   switch (getValueTypeTag()) {
-    case JITTypeTag::BOOL: {
-      if (auto const_bool = llvm::dyn_cast<llvm::ConstantInt>(llvm_rh.llvm_value_)) {
-        return const_bool->isOne() ? JITValuePointer(this) : JITValuePointer(&rh);
-      }
+    case JITTypeTag::BOOL:
+    case JITTypeTag::INT8:
+    case JITTypeTag::INT16:
+    case JITTypeTag::INT32:
+    case JITTypeTag::INT64:
+    case JITTypeTag::INT128:
```

Comment:
`return makeJITValuePointer<LLVMJITValue>(
      JITTypeTag::BOOL, parent_function_, ans, "and", false);` return type should be equal to `getValueTypeTag()`